src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs

```diff
@@ -0,0 +1,428 @@
+				poms_to_verify.Add (art);
+
+		foreach (var pom in AdditionalManifests ?? [])
+			pom_resolver.RegisterFromAndroidAdditionalJavaManifest (pom);
+
+		// If there were errors loading the requested POMs, bail
+		if (Log.HasLoggedErrors)
+			return false;
+
+		// Populate the dependency resolver with every dependency we know about
+		var resolver = new DependencyResolver (ProjectAssetsLockFile, Log);
+
+		resolver.AddAndroidLibraries (AndroidLibraries);
+		resolver.AddPackageReferences (PackageReferences);
+		resolver.AddProjectReferences (ProjectReferences);
+		resolver.AddIgnoredDependencies (IgnoredDependencies);
+
+		// Parse microsoft-packages.json so we can provide package recommendations
+		var ms_packages = new MicrosoftNuGetPackageFinder (MicrosoftPackagesFile, Log);
+
+		// Verify dependencies
+		foreach (var pom in poms_to_verify) {
+			if (TryResolveProject (pom, pom_resolver, out var resolved_pom)) {
+				foreach (var dependency in resolved_pom.Dependencies.Where (d => (d.IsRuntimeDependency () || d.IsCompileDependency ()) && !d.IsOptional ()))
+					resolver.EnsureDependencySatisfied (dependency, ms_packages);
+			}
+		}
+
+		return !Log.HasLoggedErrors;
+	}
+
+	static bool TryResolveProject (Artifact artifact, IProjectResolver resolver, [NotNullWhen (true)]out ResolvedProject? project)
+	{
+		// ResolvedProject.FromArtifact will throw if a POM cannot be resolved, but our MSBuildLoggingPomResolver
+		// has already logged the failure as an MSBuild error.  We don't want to log it again as an unhandled exception.
+		try {
+			project = ResolvedProject.FromArtifact (artifact, resolver);
+			return true;
+		} catch {
+			project = null;
+			return false;
+		}
+	}
+}
+
+class DependencyResolver
+{
+	readonly Dictionary<string, Artifact> artifacts = new ();
+
+	readonly NuGetPackageVersionFinder? finder;
+	readonly TaskLoggingHelper log;
+
+	public DependencyResolver (string? lockFile, TaskLoggingHelper log)
+	{
+		this.log = log;
+
+		if (File.Exists (lockFile))
+			finder = NuGetPackageVersionFinder.Create (lockFile!, log);
+	}
+
+	public bool EnsureDependencySatisfied (ResolvedDependency dependency, MicrosoftNuGetPackageFinder packages)
+	{
+		if (!dependency.Version.HasValue ())
+			log.LogMessage ("Could not determine required version of Java dependency '{0}:{1}'. Validation of this dependency will not take version into account.", dependency.GroupId, dependency.ArtifactId);
+
+		var satisfied = TrySatisfyDependency (dependency);
+
+		if (satisfied)
+			return true;
+
+		var suggestion = packages.GetNuGetPackage ($"{dependency.GroupId}:{dependency.ArtifactId}");
+		var artifact_spec = dependency.Version.HasValue () ? dependency.VersionedArtifactString : dependency.ArtifactString;
+
+		if (suggestion is string nuget)
+			log.LogCodedError ("XA4242", Properties.Resources.XA4242, artifact_spec, nuget);
+		else
+			log.LogCodedError ("XA4242", Properties.Resources.XA4241, artifact_spec);
```

Comment:
I think this should be `"XA4241"`, not `"XA4242"`, based on the `Properties.Resources.XA4241` property name.